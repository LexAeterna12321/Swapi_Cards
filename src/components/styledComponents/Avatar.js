import styled from "styled-components";
const Avatar = styled.section`
  width: 90%;
  margin: 0 auto;
`;

export default Avatar;
