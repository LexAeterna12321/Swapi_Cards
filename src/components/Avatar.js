import React from "react";
import PropTypes from "prop-types";
import { Avatar as StyledAvatar, Image } from "./styledComponents";

const Avatar = ({ name }) => (
  <StyledAvatar>
    {name === "C-3PO" ? (
      <Image src="./img/c3po.jpg" alt="c-3po" />
    ) : (
      <Image src="./img/luke.jpg" alt="Luke Skywalkwer" />
    )}
  </StyledAvatar>
);

Avatar.propTypes = {
  name: PropTypes.string.isRequired
};

export default Avatar;
