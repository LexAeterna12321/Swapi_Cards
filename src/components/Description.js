import React from "react";
import PropTypes from "prop-types";
import { CardDescription } from "./styledComponents";

const Description = ({ name, birth_year, gender }) => (
  <CardDescription>
    <h3>Name: {name}</h3>
    <p>Birth Date: {birth_year}</p>
    <p>Gender: {gender}</p>
  </CardDescription>
);

Description.propTypes = {
  name: PropTypes.string.isRequired,
  birth_year: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired
};

export default Description;
