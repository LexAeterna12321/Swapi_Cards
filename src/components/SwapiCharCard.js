import React from "react";
import PropTypes from "prop-types";
import { Card } from "./styledComponents";

const SwapiCharCard = ({ children }) => <Card>{children}</Card>;

SwapiCharCard.propTypes = {
  Avatar: PropTypes.element,
  Description: PropTypes.element
};

export default SwapiCharCard;
