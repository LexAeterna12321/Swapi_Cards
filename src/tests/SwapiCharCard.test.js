import React from "react";
import SwapiCharCard from "../components/SwapiCharCard";
import Avatar from "../components/Avatar";
import Description from "../components/Description";
import { shallow } from "enzyme";

it("expects to render SwapiCharCard component", () => {
  expect(shallow(<SwapiCharCard />).length).toEqual(1);
  expect(shallow(<SwapiCharCard />)).toMatchSnapshot();
});

it("renders children when passed in", () => {
  const wrapper = shallow(
    <SwapiCharCard>
      <div className="cards__card">
        <Avatar />
        <Description />
      </div>
    </SwapiCharCard>
  );
  expect(wrapper.contains(<Avatar />)).toEqual(true);
  expect(wrapper.contains(<Description />)).toEqual(true);
});
