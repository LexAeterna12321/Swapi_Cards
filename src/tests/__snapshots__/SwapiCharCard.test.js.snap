// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render SwapiCharCard component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SwapiCharCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "rules": Array [
          "
  padding: 20px 0;
  width: 70%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  padding: 20px 0;
  width: 70%;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
